src/test/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtilsUnitTest.java

```diff
@@ -93,6 +97,48 @@ public void testHomozygousAlleleList() throws Exception {
         Assert.assertEquals(alleles, Arrays.asList(T, T));
     }
 
+    final static private Locatable START_AT_1 = new SimpleInterval("1",1,1);
+    final static private Locatable START_AT_2 = new SimpleInterval("1",2,2);
+
+    @DataProvider
+    Object[][] provideDataForDetermineReferenceAllele() {
+        // {list of vcs' allele lists, list of vc starts, Locatable for this start, expected ref allele
+        return new Object[][] {
+                { Collections.emptyList(), Collections.emptyList(), null, null },
+                { Collections.emptyList(), Collections.emptyList(), START_AT_1, null },
+                { Arrays.asList(Arrays.asList(Aref)), Arrays.asList(1), START_AT_1, Aref },
+                { Arrays.asList(Arrays.asList(Aref)), Arrays.asList(1), START_AT_2, null },
+                { Arrays.asList(Arrays.asList(Aref), Arrays.asList(ATref)), Arrays.asList(1,1), START_AT_1, ATref },
+                { Arrays.asList(Arrays.asList(ATref), Arrays.asList(Aref)), Arrays.asList(1,1), START_AT_1, ATref },
+                { Arrays.asList(Arrays.asList(Aref), Arrays.asList(ATref)), Arrays.asList(1,2), START_AT_1, Aref },
+                { Arrays.asList(Arrays.asList(Aref), Arrays.asList(ATref)), Arrays.asList(1,1), START_AT_2, null },
+                { Arrays.asList(Arrays.asList(Aref), Arrays.asList(ATref)), Arrays.asList(1,1), null, ATref },
+                { Arrays.asList(Arrays.asList(Aref, C), Arrays.asList(ATref, ATCATC)), Arrays.asList(1,1), START_AT_1, ATref },
+                { Arrays.asList(Arrays.asList(Aref), Arrays.asList(ATCref), Arrays.asList(ATref)), Arrays.asList(1,1,1), START_AT_1, ATCref }
+        };
+    }
+
+    @Test(dataProvider = "provideDataForDetermineReferenceAllele")
+    public void testDetermineReferenceAllele(final List<List<Allele>> alleleLists, final List<Integer> starts, final Locatable loc, final Allele expectedRef) {
+        final List<VariantContext> vcs = IntStream.range(0, alleleLists.size())
+                .mapToObj(n -> {
+                    final int start = starts.get(n);
+                    final int stop = starts.get(n) + alleleLists.get(n).get(0).length() - 1;
+                    return new VariantContextBuilder(Integer.toString(n), "1", start, stop, alleleLists.get(n)).make();
+                })
+                .collect(Collectors.toList());
+
+        final Allele ref = GATKVariantContextUtils.determineReferenceAllele(vcs, loc);
+
+        if (expectedRef == null) {
+            Assert.assertTrue(ref == null);
+        } else {
+            Assert.assertTrue(ref.isReference());
+            Assert.assertEquals(ref.getBaseString(), expectedRef.getBaseString());
+        }
+        Assert.assertTrue(ref == expectedRef || ref.getBaseString().equals(expectedRef.getBaseString()));
```

Comment:
This last assert doesn't seem necessary since you already check for allele equality above.

Reply:
agreed and done